lib/typeprof/core/graph/box.rb

```diff
@@ -519,9 +523,12 @@ def pass_arguments(changes, genv, a_args)
 
         if start_rest < end_rest
           if @f_args.rest_positionals
-            f_arg = @f_args.rest_positionals
             (start_rest..end_rest-1).each do |i|
-              changes.add_edge(genv, a_args.positionals[i], f_arg)
+              @f_args.rest_positionals.each_type do |ty|
+                if ty.is_a?(Type::Instance) && ty.mod == genv.mod_ary && ty.args[0]
```

Comment:
I noticed some duplicate code, which feels a bit dirty, but I decided to go ahead and open the PR since cleaning it up won't matter if the approach itself is flawed. 
Do you have any suggestions for making this part of the code more DRY? The simplest solution would be to define a private method to consolidate the logic, but it doesn't feel like the optimal approach...